base/agda/Base/Free.agda

```diff
@@ -1,5 +1,16 @@
 module Base.Free where
 
+-- for transformation
+open import Function            using (case_of_)      public
+open import Data.Bool           using (if_then_else_) public
+open import Size                using (Size; ↑_)      public
+
+-- for proofs
+open import Base.Extensionality using (ext)           public
+import Relation.Binary.PropositionalEquality as Eq
+open Eq             public
+open Eq.≡-Reasoning public
```

Comment:
I think that we should export only functions and types that are needed by the generated code. The users can import libraries for their proofs themselves.